horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/TestServiceImpl.java

```diff
@@ -636,9 +636,6 @@ public void recalculateDatasets(int testId) {
         RecalculationStatus status = new RecalculationStatus(RunDAO.count("testid = ?1 AND trashed = false", testId));
         // we don't have to care about races with new runs
         RecalculationStatus prev = recalculations.putIfAbsent(testId, status);
-        // ensure the recalculation is removed, with this approach we should guarantee
-        // it gets removed even if transaction-level exception occurs, e.g., timeout
-        Util.registerTxSynchronization(tm, txStatus -> recalculations.remove(testId, status));
```

Comment:
@johnaohara I also had to remove this because the run transformation is executed async making use of `mediator.executeBlocking(..` therefore this `recalculations.remove` was getting called immediately before returning this method but much before the recalculation was completed which is wrong.. and it does **not** prevent multiple recalculation on the same test to happen concurrently

Reply:
I think we should be fairly safe even without that check as we are wrapping the `mediator.transform` with try-finally that should ensure the finally clause to get called even if transaction-level exceptions are thrown:
```java
                mediator.executeBlocking(() -> {
                    int newDatasets = 0;
                    try {
                        newDatasets = mediator.transform(runId, true);
                    } finally {
                        synchronized (status) {
                            status.finished++;
                            status.datasets += newDatasets;
                            if (status.finished == status.totalRuns) {
                                recalculations.remove(testId, status);
                            }
                        }
                    }
                });
```